SpotifyUserLogin/src/main/java/useCase/SurpriseMeInteractor.java

```diff
@@ -0,0 +1,48 @@
+package useCase;
+
+import api.SpotifyInteractor;
+import entities.Song;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.util.Random;
+
+public class SurpriseMeInteractor {
+    private final SpotifyInteractor spotifyInteractor;
+
+    public SurpriseMeInteractor(SpotifyInteractor spotifyInteractor) {
+        this.spotifyInteractor = spotifyInteractor;
+    }
+
+    public Song getRandomSongFromFriends(boolean usePreferences) {
+        // Retrieve playlists that are not owned by the user (friends' playlists)
+        JSONObject playlistsJson = spotifyInteractor.getNonOwnedPlaylists(20, 0);
+
+        JSONArray playlists = playlistsJson.optJSONArray("items");
+
+        if (playlists == null || playlists.isEmpty()) {
+            throw new RuntimeException("No friend playlists available.");
+        }
+
+        // Pick a random playlist
+        Random random = new Random();
+        JSONObject randomPlaylist = playlists.getJSONObject(random.nextInt(playlists.length()));
+
+        // Fetch items (songs) from the playlist
+        JSONObject playlistItemsJson = spotifyInteractor.getPlaylistItems(randomPlaylist.getString("id"), 50, 0);
+        JSONArray tracks = playlistItemsJson.optJSONArray("items");
+
+        if (tracks == null || tracks.isEmpty()) {
```

Comment:
There's a little bit of duplicate code from line 23. There's not enough time to go about fixing it, but it's worth thinking about how this can be fixed